About Dataset

Context

Sales analytics is the practice of generating insights from sales data, trends, and metrics to set targets and forecast future sales performance. Sales analysis is mining your data to evaluate the performance of your sales team against its goals. It provides insights about the top performing and underperforming products/services, the problems in selling and market opportunities, sales forecasting, and sales activities that generate revenue.

Content

Order ID - An Order ID is the number system that Amazon uses exclusively to keep track of orders. Each order receives its own Order ID that will not be duplicated. This number can be useful to the seller when attempting to find out certain details about an order such as shipment date or status.

Product - The product that have been sold.

Quantity Ordered - Ordered Quantity is the total item quantity ordered in the initial order (without any changes).

Price Each - The price of each products.

Order Date - This is the date the customer is requesting the order be shipped.

Purchase Address - The purchase order is prepared by the buyer, often through a purchasing department. The purchase order, or PO, usually includes a PO number, which is useful in matching shipments with purchases; a shipping date; billing address; shipping address; and the request items, quantities and price.
Target

A target market analysis is an assessment of how your product or service fits into a specific market and where it will gain the most.

Task:

Q: What was the best month for sales? How much was earned that month?

Q: What City had the highest number of sales?

Q: What time should we display adverstisement to maximize likelihood of customer's buying product?

Q: What product sold the most? Why do you think it sold the most?

In [8]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import glob
# might be needed depending on your version of Jupyter
%matplotlib inline

In [9]:

# Find all CSV files in the current directory
files = glob.glob("*.csv")

# Create an empty list to store the dataframes
dfs = []

# Loop through each file, read it into a dataframe, and append it to the list
for file in files:
    df = pd.read_csv(file)
    dfs.append(df)

# Concatenate the dataframes into a single dataframe
df = pd.concat(dfs, ignore_index=True)

# Write the combined dataframe to a new CSV file
df.to_csv("combined_data.csv", index=False)

In [10]:
df.head()

Order ID                     Product Quantity Ordered Price Each  \
0   176558        USB-C Charging Cable                2      11.95   
1      NaN                         NaN              NaN        NaN   
2   176559  Bose SoundSport Headphones                1      99.99   
3   176560                Google Phone                1        600   
4   176560            Wired Headphones                1      11.99   

       Order Date                      Purchase Address  
0  04/19/19 08:46          917 1st St, Dallas, TX 75001  
1             NaN                                   NaN  
2  04/07/19 22:30     682 Chestnut St, Boston, MA 02215  
3  04/12/19 14:38  669 Spruce St, Los Angeles, CA 90001  
4  04/12/19 14:38  669 Spruce St, Los Angeles, CA 90001

In [11]:
# let's drop the rows of NaN data!
df =df.dropna(how='all')

# okay, let's check it again!
"NaN Value:"
df[df.isna().any(axis=1)]

# future warning! ValueError: invalid literal for int() with base 10: 'Or'
"Clean Future Warnings:"
df = df[df['Order Date'].str[0:2] != 'Or']
df

Order ID                     Product Quantity Ordered Price Each  \
0        176558        USB-C Charging Cable                2      11.95   
2        176559  Bose SoundSport Headphones                1      99.99   
3        176560                Google Phone                1        600   
4        176560            Wired Headphones                1      11.99   
5        176561            Wired Headphones                1      11.99   
...         ...                         ...              ...        ...   
747395   259353      AAA Batteries (4-pack)                3       2.99   
747396   259354                      iPhone                1        700   
747397   259355                      iPhone                1        700   
747398   259356      34in Ultrawide Monitor                1     379.99   
747399   259357        USB-C Charging Cable                1      11.95   

            Order Date                         Purchase Address  
0       04/19/19 08:46             917 1st St, Dallas, TX 75001  
2       04/07/19 22:30        682 Chestnut St, Boston, MA 02215  
3       04/12/19 14:38     669 Spruce St, Los Angeles, CA 90001  
4       04/12/19 14:38     669 Spruce St, Los Angeles, CA 90001  
5       04/30/19 09:27        333 8th St, Los Angeles, CA 90001  
...                ...                                      ...  
747395  09/17/19 20:56   840 Highland St, Los Angeles, CA 90001  
747396  09/01/19 16:00  216 Dogwood St, San Francisco, CA 94016  
747397  09/23/19 07:39     220 12th St, San Francisco, CA 94016  
747398  09/19/19 17:30   511 Forest St, San Francisco, CA 94016  
747399  09/30/19 00:18   250 Meadow St, San Francisco, CA 94016  

[743800 rows x 6 columns]

 What was the best month for sales? How much was earned that month?

In [12]:
df['Order Date'] = pd.to_datetime(df['Order Date'])

In [13]:
df.head()

Order ID                     Product Quantity Ordered Price Each  \
0   176558        USB-C Charging Cable                2      11.95   
2   176559  Bose SoundSport Headphones                1      99.99   
3   176560                Google Phone                1        600   
4   176560            Wired Headphones                1      11.99   
5   176561            Wired Headphones                1      11.99   

           Order Date                      Purchase Address  
0 2019-04-19 08:46:00          917 1st St, Dallas, TX 75001  
2 2019-04-07 22:30:00     682 Chestnut St, Boston, MA 02215  
3 2019-04-12 14:38:00  669 Spruce St, Los Angeles, CA 90001  
4 2019-04-12 14:38:00  669 Spruce St, Los Angeles, CA 90001  
5 2019-04-30 09:27:00     333 8th St, Los Angeles, CA 90001

In [18]:
df['month'] = df['Order Date'].dt.month


In [20]:
monthly_sales = df.groupby('month')['Price Each'].sum()

In [22]:
df['Price Each'].dtype

dtype('O')

In [23]:
df['Price Each'] = pd.to_numeric(df['Price Each'], errors='coerce')

In [24]:
best_month= df.groupby('month')['Price Each'].sum().idxmax()

In [25]:
print("The best month for sales was", best_month)

The best month for sales was 12


How much was earned that month?


In [27]:
df['Year'] = df['Order Date'].dt.year


In [29]:
# Filter the data for the month of December
dec_sales = df.loc[(df['month'] == 12), 'Price Each']


In [30]:
dec_total_sales = dec_sales.sum()

In [31]:

# Print the result
print("Total sales for December were ${:,.2f}.".format(dec_total_sales))

Total sales for December were $18,353,661.64.


What City had the highest number of sales?

In [33]:
# Extract the city from the 'Purchase Address' column and create a new column called 'City'
df['City'] = df['Purchase Address'].apply(lambda x: x.split(',')[1])


In [35]:
# Calculate the total sales for each city
city_sales = df.groupby('City')['Price Each'].sum()

In [39]:
# Find the city with the highest sales
highest_sales_city = city_sales.idxmax()

In [40]:
print("The city with the highest sales was {} with total sales of ${:,.2f}.".format(highest_sales_city, city_sales[highest_sales_city]))

The city with the highest sales was  San Francisco with total sales of $32,845,846.96.


What time should we display adverstisement to maximize likelihood of customer's buying product?

In [41]:
# Create a new column 'Hour' to represent the hour of the day when the order was made
df['Hour'] = df['Order Date'].dt.hour

In [42]:
# Count the number of orders for each hour of the day
hourly_sales = df.groupby('Hour')['Quantity Ordered'].sum()


In [46]:
hourly_sales = hourly_sales.astype('float64')

In [55]:
# Find the hour with the highest number of orders
best_time = hourly_sales.idxmax()

In [56]:
# Print the result
print("The best time to display advertisements is around {}ockloc.".format(best_time))

The best time to display advertisements is around 0ockloc.


What products are most often sold together?

In [62]:
# Convert Quantity Ordered column to numeric data type
df['Quantity Ordered'] = pd.to_numeric(df['Quantity Ordered'])

In [63]:
# Group by Product and sum the Quantity Ordered column
product_group = df.groupby('Product')['Quantity Ordered'].sum()


In [64]:
# Find the product with the maximum Quantity Ordered
most_sold_product = product_group.idxmax()


In [65]:
print("The most sold product was:", most_sold_product)

The most sold product was: AAA Batteries (4-pack)


Conclusion

The best month for sales was 12, Total sales for December were $18,353,661.64, The city with the highest sales was  San Francisco with total sales of $32,845,846.96. The best time to display advertisements is around 00:00 ockloc and The most sold product was: AAA Batteries (4-pack)
